src/components/Match/Overview/Timeline.jsx

```diff
@@ -0,0 +1,302 @@
+                        />
+                      }
+                      <ReactTooltip
+                        id={`event_${i}`}
+                        effect="solid"
+                        place="right"
+                      >
+                        {obj.type === 'firstblood' &&
+                          <section>
+                            {match.players
+                              .filter(player => player.player_slot === obj.player_slot)
+                              .map(player => (
+                                <aside style={{ color: playerColors[obj.player_slot] }}>
+                                  <img
+                                    src={`${API_HOST}/apps/dota2/images/heroes/${heroes[player.hero_id].name.split('npc_dota_hero_')[1]}_icon.png`}
+                                    role="presentation"
+                                  />
+                                  {player.name || player.personaname || strings.general_anonymous}
+                                </aside>
+                              ))
+                            }
+                            <span>
+                              {obj.key ? strings.timeline_firstblood_key : strings.timeline_firstblood}
+                            </span>
+                            {obj.key &&
+                              <aside>
+                                <img
+                                  src={`${API_HOST}/apps/dota2/images/heroes/${obj.key.split('npc_dota_hero_')[1]}_icon.png`}
+                                  role="presentation"
+                                />
+                                {match.players
+                                  .filter(player =>
+                                    player.hero_id === heroes[Object.keys(heroes).filter(key => heroes[key].name === obj.key)].id,
+                                  ).map(player => (
+                                    <div style={{ color: playerColors[player.player_slot] }}>
+                                      {player.name || player.personaname || strings.general_anonymous}
+                                    </div>
+                                  ))
+                                }
+                              </aside>
+                            }
+                          </section>
+                        }
+                        {obj.type === 'roshan' &&
+                          match.players
+                            .filter(player => player.player_slot === aegis[0][obj.key].player_slot)
+                            .map(player => (
+                              <section key={i}>
+                                <aside style={{ color: playerColors[player.player_slot] }}>
+                                  <img
+                                    src={`${API_HOST}/apps/dota2/images/heroes/${heroes[player.hero_id].name.split('npc_dota_hero_')[1]}_icon.png`}
```

Comment:
we really should move this into some utility function, we're doing it all over the place (several places in match columns)